File: base/src/main/vsdk/toolkit/common/dataStructures/_CircularDoubleLinkedListNode.java
package vsdk.toolkit.common.dataStructures;

import java.io.Serial;
import vsdk.toolkit.common.FundamentalEntity;

public class _CircularDoubleLinkedListNode<E> extends FundamentalEntity
{
    @Serial private static final long serialVersionUID = 20070422L;

    public E data;
    public _CircularDoubleLinkedListNode<E> next;
    public _CircularDoubleLinkedListNode<E> previous;
}

File: typescript/base/src/main/vsdk/toolkit/common/dataStructures/_CircularDoubleLinkedListNode.ts
import { FundamentalEntity } from "../FundamentalEntity.js";

/** Link node used by CircularDoubleLinkedList. */
export class _CircularDoubleLinkedListNode<E> extends FundamentalEntity {
  public data: E | null = null;
  public next: _CircularDoubleLinkedListNode<E> | null = null;
  public previous: _CircularDoubleLinkedListNode<E> | null = null;
}
